src/Components/MovieTile/__tests__/MovieTile.test.js

```diff
@@ -1,47 +1,57 @@
 import { render, screen } from '@testing-library/react';
 import { MovieTile } from '../MovieTile';
 import userEvent from '@testing-library/user-event';
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
 
 describe('MovieTile', () => {
   test('renders all details', () => {
     const testData = {
+      id: '123',
       imageUrl: 'imageUrl',
       movieName: 'testName',
       releaseYear: 'testYear',
       relevantGenres: ['genre1', 'genre2'],
     };
 
-    const { asFragment } = render(<MovieTile {...testData} />);
+    const { asFragment } = render(<MovieTile {...testData} />, { wrapper: BrowserRouter });
 
     expect(asFragment()).toMatchSnapshot();
   });
 
   test('renders with required props', () => {
     const testData = {
+      id: '123',
       imageUrl: 'imageUrl',
       movieName: 'testName',
       releaseYear: 'testYear',
     };
 
-    const { asFragment } = render(<MovieTile {...testData} />);
+    const { asFragment } = render(<MovieTile {...testData} />, { wrapper: BrowserRouter });
```

Comment:
You can also create a helper function as dfescribed in https://testing-library.com/docs/example-react-router/#reducing-boilerplate.
To avoid passing `wrapper` in every `render`.